org.eclipse.jdt.debug.jdi.tests/tests/org/eclipse/debug/jdi/tests/program/MainClass.java

```diff
@@ -203,7 +203,8 @@ public void print(OutputStream out) {
 	 * @since 3.3
 	 */
 	public int printNumber(OutputStream out, int num) {
-		String blah = "foo"+foo();
+		String blah = "foo" + foo();
+		System.out.println(blah);
```

Comment:
This line is purposeful. Without this line, the test case [ForceEarlyReturnTests#testForceEarlyReturnIntType](https://ci.eclipse.org/jdt/job/eclipse.jdt.debug-github/job/PR-76/9/testReport/junit/org.eclipse.debug.jdi.tests/ForceEarlyReturnTests/testForceEarlyReturnIntType/) in CI machine always fails. The reason is that the Maven compiler (aka. javac) doesn't store the unused variable `blah` in the LocalVariableTable of the binary class, so the unit test cannot fetch this variable value from JDI.

You can use `javap -p` to compare the class files generated by ECJ and Maven. ECJ generates LocalVariableTable for all local variables, so running this unit test with Eclipse will pass. Running it from the Maven command line will fail because the variable cannot be found from the StackFrame's visible variables.

Adding the sysout line is a workaround and make `blah` not unused.

Reply:
> Running it from the Maven command line will fail because the variable cannot be found from the StackFrame's visible variables.

I would then try to fix maven arguments to preserve LVT. Because it is absolutely surprising behavior that would cause later changes fail for no obvious reason.

Reply:
@iloveeclipse Nice suggestion, thanks. I make tycho-compiler-plugin to reuse project settings for compiler configuration, it now preserves LVT for unused locals.